### DISCUSSION TITLE
How do I disable the completion results getting inlined automatically?

### DISCUSSION THREAD

**Discussion by diocletiann** [LazyVim/LazyVim#2333] [Q&A] [answered]
It covers up the text that was there and has been confusing thus far. Thanks!

**Comment by dpetka2001** [comment]
You should read the `nvim-cmp` [FAQ](https://github.com/hrsh7th/nvim-cmp/blob/538e37ba87284942c1d76ed38dd497e54e65b891/doc/cmp.txt#L884). It tries to answer several questions compared to what you want to achieve. I've only disabled the `preselect` feature myself, not tried the rest.

**Reply by diocletiann** [reply]
I skimmed it and I didn't find the setting that is actually drawing the selection output on top of the active line, before I commit the selection.

**Reply by dpetka2001** [reply]
What exactly do you want to disable? Do you want to disable the whole autocompletion menu? And then just press a key for it to come up? Or disable just the docs preview?

**Reply by diocletiann** [reply]
<img width="299" alt="image" src="https://github.com/LazyVim/LazyVim/assets/14058796/204ebfb4-dfad-4c42-9efc-e0f29f423152">


I typed `std.mem.` and then it automatically filled in the active line with the first result, and it does so as I scroll through the selection options. This is ok when there's nothing else on the line but when changing an existing line it covers up (or manipulates) the rest of the text.

**Reply by dpetka2001** [reply, marked as answer]
I believe that feature is the `ghost_text` and you can disable it with
```lua
-- ~/.config/nvim/lua/plugins/cmp.lua
return {
  {
    "hrsh7th/nvim-cmp",
    opts = {
      experimental = {
        ghost_text = false,
      },
    },
  },
}
```
Test it to see if that's what you want or if I didn't understand correctly.

**Reply by diocletiann** [reply]
hmm, that disables the completion dropdown entirely.

**Reply by dpetka2001** [reply]
No, that disables only the light grey text that appears after you type `std.mem.`. So the `window(${1:})` part won't appear after you type `std.mem.`. The dropdown menu will still be visible. Just tested it.

**Reply by dpetka2001** [reply]
This is what it looks like on my end with that setting
![2024-01-10_23-30](https://github.com/LazyVim/LazyVim/assets/12776461/d970dd31-1fda-4fe9-98bd-7068419707be)

**Reply by dpetka2001** [reply]
And this with the default LazyVim setting
![2024-01-10_23-32](https://github.com/LazyVim/LazyVim/assets/12776461/3288ca1f-17bd-4969-a576-861d01abeb5b)

**Reply by diocletiann** [reply]
I did it wrong, it works now! Thank you!

**Reply by Mazin-Ayman** [reply]
Why it didn't work for me?

**Reply by dpetka2001** [reply]
Because now the default completion plugin is `blink.cmp`, not `nvim-cmp`. You have to do
```lua
{
    "saghen/blink.cmp",
    ---@module 'blink.cmp'
    ---@type blink.cmp.Config
    opts = {
      completion = {
        ghost_text = {
          enabled = false,
        },
      },
    },
  },
```
You can also find relevant information about blink settings in [blink docs](https://cmp.saghen.dev/).
